rest/config.go

```diff
@@ -1463,8 +1463,24 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 		// inherit properties the bootstrap config
 		cnf.CACertPath = sc.Config.Bootstrap.CACertPath
 
+		// We need to check for corruption in the database config (CC. CBG-3292). If the fetched config doesn't match the
+		// bucket name we got the config from we need to maker this db context as corrupt. Then remove the context and
+		// in memory representation on the server context.
 		bucketCopy := bucket
-		cnf.Bucket = &bucketCopy
+		if bucket != *cnf.Bucket {
+			base.DebugfCtx(ctx, base.KeyConfig, "Mismatch in database config bucket name: %s and backend bucket: %s You must update database config immediately", *cnf.Bucket, bucket)
+			// track corrupt database context
+			ok := sc._addCorruptDatabase(ctx, dbName)
+			if !ok {
+				base.DebugfCtx(ctx, base.KeyConfig, "Failed ot add sb: %s to corrupt list", cnf.Name)
+			}
+			// don't load config + remove from server context (apart from corrupt database map)
+			sc._removeDatabase(ctx, cnf.Name)
```

Comment:
We need to remove this if the database exists (but it might not) so we probably want two tests and code pathways:

1. if the database config is modified and corrupted, we want to log with a really clear message that we are unloading the database. I think the return value of `_removeDatabase` will tell us if there's an active database
2. we read the database config, but it's corrupt

I guess the reason to keep track of invalid databases is to avoid log spamming for this issue, but because it's so fatal, I wonder whether that's necessary. It would log every second, so that's bad, but having no logging about this is also bad. I think the optimal log strategy might be somewhere in the middle and creating a datastructure that holds databases without bucket names or without bucket names that locks separately from other server context locks:

Prototype:

```
type invalidBucketName struct {
        lastLogTime atomic.Int64
}

type invalidBucketNameDatabases struct {
      dbNames map[string]invalidBucketNameDatabase
      m sync.RWMutex
}

func (*d invalidBucketNameDatabases) addInvalidDatabase(dbname string) {
       // lock map
       // check last logged time, if past threshold log again
       // update log time
}
```